src/bitmessagekivy/baseclass/allmail.py

```diff
@@ -0,0 +1,202 @@
+
+
+class Allmails(Screen, HelperAllMail):
+    """Allmails Screen for kivy Ui"""
+
+    data = ListProperty()
+    has_refreshed = True
+    all_mails = ListProperty()
+    account = StringProperty()
+
+    def __init__(self, *args, **kwargs):
+        """Method Parsing the address"""
+        super(Allmails, self).__init__(*args, **kwargs)
+        if state.association == '':
+            if state.kivyapp.variable_1:
+                state.association = state.kivyapp.variable_1[0]
+        Clock.schedule_once(self.init_ui, 0)
+
+    def init_ui(self, dt=0):
+        """Clock Schdule for method all mails"""
+        self.loadMessagelist()
+        logger.debug(dt)
+
+    def loadMessagelist(self):
+        """Load Inbox, Sent anf Draft list of messages"""
+        self.account = state.association
+        self.ids.tag_label.text = ''
+        self.allMessageQuery(0, 20)
+        if self.all_mails:
+            self.ids.tag_label.text = 'All Mails'
+            state.kivyapp.get_inbox_count()
+            state.kivyapp.get_sent_count()
+            state.all_count = str(
+                int(state.sent_count) + int(state.inbox_count))
+            self.set_AllmailCnt(state.all_count)
+            self.set_mdlist()
+            self.ids.scroll_y.bind(scroll_y=self.check_scroll_y)
+        else:
+            self.set_AllmailCnt('0')
+            self.ids.ml.add_widget(self.default_label_when_no_email())
+
+    def allMessageQuery(self, start_indx, end_indx):
+        """Retrieving data from inbox or sent both tables"""
+        self.all_mails = sqlQuery(
+            "SELECT toaddress, fromaddress, subject, message, folder, ackdata"
+            " As id, DATE(senttime) As actionTime, senttime as msgtime FROM sent WHERE"
+            " folder = 'sent' and fromaddress = '{0}'"
+            " UNION SELECT toaddress, fromaddress, subject, message, folder,"
+            " msgid As id, DATE(received) As actionTime, received as msgtime FROM inbox"
+            " WHERE folder = 'inbox' and toaddress = '{0}'"
+            " ORDER BY actionTime DESC limit {1}, {2}".format(
+                self.account, start_indx, end_indx))
+
+    def set_AllmailCnt(self, Count):  # pylint: disable=no-self-use
+        """This method is used to set allmails message count"""
+        allmailCnt_obj = state.kivyapp.root.ids.content_drawer.ids.allmail_cnt
+        allmailCnt_obj.ids.badge_txt.text = showLimitedCnt(int(Count))
+
+    def set_mdlist(self):
+        """This method is used to create mdList for allmaills"""
+        data_exist = len(self.ids.ml.children)
+        for item in self.all_mails:
+            body = item[3].decode() if isinstance(item[3], bytes) else item[3]
+            subject = item[2].decode() if isinstance(item[2], bytes) else item[2]
+            message_row = CutsomSwipeToDeleteItem(
+                text=item[1],
+            )
+            listItem = message_row.ids.content
+            secondary_text = (subject[:50] + '........' if len(
```

Comment:
maybe there can be a function something like `def string_max_length(max_lenght=50):`